lib/devfs/include/devfs/device.h

```diff
@@ -190,22 +219,157 @@ devtoname(struct device *dev)
+ * 	the function returns.
+ * @return
+ * 	The result of the device's open routine
+ */
 int device_open(const char *name, int mode, struct device **devp);
+
+/*
+ * Closes the device specified by dev.
+ * The function does not return an error if the target driver does not have
+ * a close routine.
+ *
+ * @param dev
+ * 	The device to be closed
+ * @return
+ * 	The result of the device's close routine
+ */
 int device_close(struct device *dev);
+
+/*
+ * Performs a scatter/gather read on the device pointed to by dev.
+ *
+ * @param dev
+ * 	The device to read from
+ * @param uio
+ * 	The structure containing the scatter-gather list
+ * @param ioflags
+ * 	The flags to be sent to the read operation
+ * @return
+ * 	The result of the read operation
+ */
 int device_read(struct device *dev, struct uio *uio, int ioflags);
+
+/*
+ * Performs a scatter/gather write on the device pointed to by dev.
+ *
+ * @param dev
+ * 	The device to write to
+ * @param uio
+ * 	The structure containing the scatter-gather list
+ * @param ioflags
+ * 	The flags to be sent to the write operation
+ * @return
+ * 	The result of the write operation
+ */
 int device_write(struct device *dev, struct uio *uio, int ioflags);
+
+/*
+ * Performs an I/O control request.
+ *
+ * @param dev
+ * 	The device to perform the request on
+ * @param cmd
+ * 	The device dependent command to be executed
+ * @param arg
+ * 	The user buffer sent as argument to the command. The buffer
+ * 	must be validated by the ioctl routine of the driver.
+ * @return
+ * 	The result of the I/O control request
+ */
 int device_ioctl(struct device *dev, unsigned long cmd, void *arg);
+
+/*
+ * Returns information about a device.
+ * The target of the query should be placed in the cookie field
+ * of the parameter.
+ *
+ * @param info
+ * 	Used as a return parameter. The device must be specified
+ * 	in the cookie field and on return the id, flags and name
+ * 	fields will be filled with the requested information.
+ * @return
+ * 	0 on success and ESRCH otherwise
+ */
 int device_info(struct devinfo *info);
 
 int bdev_read(struct device *dev, struct uio *uio, int ioflags);
 int bdev_write(struct device *dev, struct uio *uio, int ioflags);
 
-int	enodev(void);
-int	nullop(void);
+/*
+ * Always returns ENODEV.
+ */
+int enodev(void);
 
+/*
+ * Always returns 0.
+ */
+int nullop(void);
+
+/*
+ * Creates a new device object.
+ *
+ * @param drv
+ * 	The driver to create the device
+ * @param name
+ * 	The name of the new device
+ * @param flags
+ * 	The flags used for the creation of the device (D_* flags: D_CHR, D_BLK,
+ * 	D_REM, D_TTY)
+ * @return
+ * 	A pointer to a device structure corresponding to the newly created
+ * 	device on success or NULL otherwise
+ */
 struct device *device_create(struct driver *drv, const char *name, int flags);
+/*
+ * Destroys a device.
+ *
+ * @param dev
+ * 	The device to be destroyed
+ * @return
+ * 	0 on success and ENODEV otherwise
```

Comment:
Please use the coding convention using lists with bullets. You can see an example at [1]

[1] https://github.com/unikraft/unikraft/blob/staging/lib/vfscore/vfs.h#L304